### PR TITLE
[M] Removed setting encoding on FileConfiguration implementation (ENT-5322)

### PR DESCRIPTION
**Pull request by Ceiu**
- Removed the getEncoding and setEncoding methods from
  FileConfiguration and all implementation classes, as it could
  not be applied in all cases, and could be lost or desync'd in
  some cases
- Added new methods to FileConfiguration for specifying the
  character encoding during config load

### CODE REVIEW

**Comment by Ceiu** [comment]
This will need to be backported to 4.0 and 4.1 upon acceptance.

**Review by JoshAlbrecht23 — CHANGES_REQUESTED** [review]
The changes look good, but I had a question on some assertions on a couple tests.

File: src/test/java/org/candlepin/config/PropertiesFileConfigurationTest.java [hunk 1]
```diff
@@ -14,105 +14,230 @@
...
-        Reader r = new FileReader(file);
+    public void testLoadFileByNameWithSpecificCharset() throws Exception {
+        PropertiesFileConfiguration config = new PropertiesFileConfiguration();
+        String filename = this.getResourceAsFile(UTF8_FILE);
 
-        config.setEncoding(UTF8);
-        config.load(r);
+        config.load(filename, UTF_8);
+        assertEquals("Mötley Crüe", config.getString("glam.band"));
+        assertNotEquals(new String("Mötley Crüe".getBytes(UTF_8), US_ASCII), config.getString("glam.band"));
```
[the remark this answers is not in the document]
Comment (Ceiu) [reply src/test/java/org/candlepin/config/PropertiesFileConfigurationTest.java:80]: So the intent here is to re-encode a string we know has unicode characters into ASCII so we can have the mangled string available to use as comparison. Providing UTF-8 for the second param during string creation would re-create the original string, effectively being a very long no-op, and would be equivalent to the assertion above it.

The previous version of the test intended to work this way, but called the no-param variant of `getBytes()`, which uses the system default charset. This is fine in cases where the system default was UTF-8, as it was for us up until about two weeks ago, but fails in cases where the default anything else.

[the remark this answers is not in the document]
Comment (JoshAlbrecht23) [reply src/test/java/org/candlepin/config/PropertiesFileConfigurationTest.java:80]: Okay, that makes sense! I appreciate the explanation.

**Review by JoshAlbrecht23 — APPROVED** [review]

File: src/main/java/org/candlepin/config/PropertiesFileConfiguration.java [hunk 2]
```diff
@@ -35,46 +35,15 @@
 public class PropertiesFileConfiguration extends AbstractConfiguration
     implements FileConfiguration {
 
-    protected Charset encoding;
+    /** Default character set to use when none is provided */
+    private static final Charset DEFAULT_CHARSET = StandardCharsets.UTF_8;
 
+    // TODO: Flatten this. We don't utilize the value we get out of having a separate configuration
+    // instance backing this fascade.
```
Comment (wottop) [inline src/main/java/org/candlepin/config/PropertiesFileConfiguration.java:42]: "facade"

**Review by wottop — CHANGES_REQUESTED** [review]

**Pull request merged**